tests/std/tests/P1222R4_flat_set/test.cpp

```diff
@@ -305,6 +357,32 @@ void test_insert_transparent() {
     assert(detector.converted);
 }
 
+void test_insert_transparent_partially_inconsistent() {
+    // `broken_key` makes inconsistency between transparent searching and conversion.
+    // however, the standard's current specification (per N4958 [flat.set.modifiers]/2) is unable to prohibit types like
+    // this.
+    struct broken_key {
+        int key;
+        explicit operator int() const {
+            return key - 10;
+        }
+    };

```

Comment:
I really hope to see cases like this being totally prohibited by the standard one day :|